src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php

```diff
@@ -65,11 +70,33 @@ public function onKernelResponse(FilterResponseEvent $event)
         $response = $event->getResponse();
         $request = $event->getRequest();
 
-        if ($response->headers->has('X-Debug-Token') && null !== $this->urlGenerator) {
+        $hasProfile = $this->profileStack instanceof ProfileStack ? $this->profileStack->has($request) : $response->headers->has('X-Debug-Token');
+
+        if ($hasProfile && null !== $this->urlGenerator) {
+            if ($this->profileStack instanceof ProfileStack) {
+                $profile = $this->profileStack->get($request);
+
+                $token = $profile->getToken();
+
+                foreach ($profile->getCollectors() as $collector) {
+                    if ($collector instanceof ExceptionDataCollector && $collector->hasException()) {
+                        $panel = $collector->getName();
+
+                        break;
+                    } elseif ($collector instanceof DumpDataCollector && $collector->getDumpsCount() > 0) {
```

Comment:
`if` is enough, as the previous `if` stops execution of this branch of the code with `break` already.